src/game/game_state.cpp

```diff
@@ -50,7 +60,13 @@ void GameState::update() {
     scene_manager.get_current_scene()->update();
 
     if (curr_phase == &dungeon_phase) {
```

Comment:
Go ahead and modify it, I plan on working in a way that minimally disrupts the existing code. Specifically:

- Make `graphical_setup` set a flag that informs `GameState` whether or not it is client or server
- Modify all phase transitions to be based on the proto enum, and demux to client or server specific phase in `set_phase`

If you do add `Context`, please pass it in through the constructor and make it protected so all derived classes have equal access to it.